hub/core/tensor.py

```diff
@@ -50,6 +56,28 @@ def create_tensor(
     IndexMeta.create(key, storage)
 
 
+def delete_tensor(key: str, storage: StorageProvider):
+    """If a tensor exists, permanently delete it.
+
+    WARNING:
+        All data inside the tensor at `key` will be deleted -- after doing this the data will be unrecoverable!
+
+    Args:
+        key (str): Key where the tensor to delete is located relative to `storage`'s root.
+        storage (StorageProvider): StorageProvider that all tensor data exists.
+
+    Raises:
+        TensorDoesNotExistError: Cannot delete a tensor that hasn't been created. Use `create_tensor`.
+    """
+
+    if not tensor_exists(key, storage):
+        raise TensorDoesNotExistError(key)
+
+    del storage[get_tensor_meta_key(key)]
+    del storage[get_index_meta_key(key)]
+    del storage[get_dataset_meta_key()]
```

Comment:
oh no.. this will delete dataset meta!

TODO: write tests that ACTUALLY check if tensor data exists (chunks!)